Spring Data JPA - ContainingIgnoreCase is not working

**Overview**
Seems that not all keywords (or combination between them) from [JPA Query Creation methods](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation) are supported.
**Test Project**
https://github.com/tpenakov/otaibe-quarkus-jpa-tests
**Describe the bug**
- Create a simple project using `quarkus-spring-data-jpa` module
- Create an `@Entity` POJO with string field in it (`name` for example)
- Create a repository extending `JpaRepository`
- Add method to the repository `List<AnyPojo> findByNameContainingIgnoreCase(String partialName)`
- Try to start the project

**Expected behavior**
Project to start normally 

**Actual behavior**
Exception is raised that the method/field `NameContaining` is not found

**Additional Notes**
This wasn't tested, but I suspect that will not going to work either: `List<AnyPojo> findByNameStartingWithIgnoreCase(String partialName)`

Hi @geoand ,
This is my error:
```
Caused by: io.quarkus.spring.data.deployment.UnableToParseMethodException: Entity org.otaibe.commons.location.domain.Location does not contain a field named: NameContaining. Offending method is findByNameContainingIgnoreCase
	at io.quarkus.spring.data.deployment.MethodNameParser.parse(MethodNameParser.java:210)
	at io.quarkus.spring.data.deployment.generate.DerivedMethodsAdder.add(DerivedMethodsAdder.java:77)
	at io.quarkus.spring.data.deployment.generate.SpringDataRepositoryCreator.implementCrudRepository(SpringDataRepositoryCreator.java:109)
	at io.quarkus.spring.data.deployment.SpringDataJPAProcessor.implementCrudRepositories(SpringDataJPAProcessor.java:120)
	at io.quarkus.spring.data.deployment.SpringDataJPAProcessor.build(SpringDataJPAProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```
This is my repository:
```
public interface LocationRepository extends JpaRepository<Location, Long> {
    /**
     * @param partialName the partial location name
     * @return the location found
     */
    List<Location> findByNameContainingIgnoreCase(String partialName);
    List<Location> findByNameStartingWithIgnoreCase(String partialName);
}
```
Anyway - I will try to create a sample project.

--- a/extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java
+++ b/extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java
@@ -250,14 +250,13 @@ public class MethodNameParser {
                 where.append(containsAnd ? " AND " : " OR ");
             }
 
+            String upperPrefix = (ignoreCase || allIgnoreCase) ? "UPPER(" : "";
+            String upperSuffix = (ignoreCase || allIgnoreCase) ? ")" : "";
+            where.append(upperPrefix).append(fieldName).append(upperSuffix);
             if ((operation == null) || "Equals".equals(operation) || "Is".equals(operation)) {
-                String upperPrefix = (ignoreCase || allIgnoreCase) ? "UPPER(" : "";
-                String upperSuffix = (ignoreCase || allIgnoreCase) ? ")" : "";
-                where.append(upperPrefix).append(fieldName).append(upperSuffix);
                 paramsCount++;
                 where.append(" = ").append(upperPrefix).append("?").append(paramsCount).append(upperSuffix);
             } else {
-                where.append(fieldName);
                 switch (operation) {
                     case "IsNot":
                     case "Not":
@@ -318,19 +317,22 @@ public class MethodNameParser {
                     case "StartingWith":
                     case "StartsWith":
                         paramsCount++;
-                        where.append(" LIKE CONCAT(?").append(paramsCount).append(", '%')");
+                        where.append(" LIKE CONCAT(").append(upperPrefix).append("?").append(paramsCount).append(upperSuffix)
+                                .append(", '%')");
                         break;
                     case "IsEndingWith":
                     case "EndingWith":
                     case "EndsWith":
                         paramsCount++;
-                        where.append(" LIKE CONCAT('%', ?").append(paramsCount).append(")");
+                        where.append(" LIKE CONCAT('%', ").append(upperPrefix).append("?").append(paramsCount)
+                                .append(upperSuffix).append(")");
                         break;
                     case "IsContaining":
                     case "Containing":
                     case "Contains":
                         paramsCount++;
-                        where.append(" LIKE CONCAT('%', ?").append(paramsCount).append(", '%')");
+                        where.append(" LIKE CONCAT('%', ").append(upperPrefix).append("?").append(paramsCount)
+                                .append(upperSuffix).append(", '%')");
                         break;
                     case "True":
                     case "False":

--- a/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/BookRepository.java
+++ b/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/BookRepository.java
@@ -18,6 +18,10 @@ public interface BookRepository extends Repository<Book, Integer> {
 
     List<Book> findByName(String name);
 
+    List<Book> findByNameContainingIgnoreCase(String name);
+
+    long countByNameStartsWithIgnoreCase(String name);
+
     boolean existsByBid(Integer id);
 
     boolean existsBookByPublicationYearBetween(Integer start, Integer end);

--- a/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/BookResource.java
+++ b/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/BookResource.java
@@ -53,6 +53,19 @@ public class BookResource {
     }
 
     @GET
+    @Path("/name/{name}/contains")
+    @Produces("application/json")
+    public List<Book> byNameContainingIgnoreCase(@PathParam("name") String name) {
+        return bookRepository.findByNameContainingIgnoreCase(name);
+    }
+
+    @GET
+    @Path("/name/{name}/count/like")
+    public long countByNameStartsWithIgnoreCase(@PathParam("name") String name) {
+        return bookRepository.countByNameStartsWithIgnoreCase(name);
+    }
+
+    @GET
     @Path("/year/{year}")
     @Produces("application/json")
     public Response findByPublicationYear(@PathParam("year") Integer year) {

--- a/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/CatRepository.java
+++ b/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/CatRepository.java
@@ -30,6 +30,8 @@ public interface CatRepository extends CrudRepository<Cat, Long> {
 
     Long countByColorIgnoreCase(String color);
 
+    Long countByColorContainsIgnoreCase(String color);
+
     Boolean existsByColorStartingWith(String color);
 
     long removeCatsByColor(String color);

--- a/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/CatResource.java
+++ b/integration-tests/spring-data-jpa/src/main/java/io/quarkus/it/spring/data/jpa/CatResource.java
@@ -89,6 +89,12 @@ public class CatResource {
     }
 
     @GET
+    @Path("/count/by/color/{color}/contains")
+    public Long countByColorContains(@PathParam("color") String color) {
+        return catRepository.countByColorContainsIgnoreCase(color);
+    }
+
+    @GET
     @Path("/exists/by/colorStartsWith/{color}")
     public Boolean existsByColorStartsWith(@PathParam("color") String color) {
         return catRepository.existsByColorStartingWith(color);

--- a/integration-tests/spring-data-jpa/src/test/java/io/quarkus/it/spring/data/jpa/BookResourceTest.java
+++ b/integration-tests/spring-data-jpa/src/test/java/io/quarkus/it/spring/data/jpa/BookResourceTest.java
@@ -74,6 +74,20 @@ public class BookResourceTest {
     }
 
     @Test
+    void testByNameContainingIgnoreCase() {
+        when().get("/book/name/FUL/contains").then()
+                .statusCode(200)
+                .body(containsString("Factfulness"));
+    }
+
+    @Test
+    void testCountByNameStartsWithIgnoreCase() {
+        when().get("/book/name/t/count/like").then()
+                .statusCode(200)
+                .body(is("1"));
+    }
+
+    @Test
     void testByYear() {
         when().get("/book/year/2012").then()
                 .statusCode(200)

--- a/integration-tests/spring-data-jpa/src/test/java/io/quarkus/it/spring/data/jpa/CatResourceTest.java
+++ b/integration-tests/spring-data-jpa/src/test/java/io/quarkus/it/spring/data/jpa/CatResourceTest.java
@@ -170,6 +170,21 @@ public class CatResourceTest {
     }
 
     @Test
+    void testCountByColorContains() {
+        when().get("/cat/count/by/color/Dummy/contains").then()
+                .statusCode(200)
+                .body(is("0"));
+
+        when().get("/cat/count/by/color/hit/contains").then()
+                .statusCode(200)
+                .body(is("1"));
+
+        when().get("/cat/count/by/color/Gre/contains").then()
+                .statusCode(200)
+                .body(is("2"));
+    }
+
+    @Test
     void testExistsByColorStartsWith() {
         when().get("/cat/exists/by/colorStartsWith/Dum").then()
                 .statusCode(200)

